File: backend/routes/recommendation.py
from flask import Blueprint, request, jsonify
from models.extracts import Extract
import random

recommendation_bp = Blueprint('recommendation', __name__)

@recommendation_bp.route('/suggest', methods=['POST'])
def suggest():
    data = request.json
    soil_type = data['soil_type']
    crop_variety = data['crop_variety']
    recommendations = Extract.query.filter_by(region=data['region']).all()
    response = [{"name": rec.name, "dosage": rec.dosage, "application_method": rec.application_method} for rec in recommendations]
    return jsonify(response), 200

File: frontend/src/components/Recommendation.js
import React, { useState } from 'react';
import axios from 'axios';

const Recommendation = () => {
  const [soilType, setSoilType] = useState('');
  const [cropVariety, setCropVariety] = useState('');
  const [region, setRegion] = useState('');
  const [recommendations, setRecommendations] = useState([]);

  const getRecommendations = () => {
    axios.post('/api/recommendation/suggest', { soil_type: soilType, crop_variety: cropVariety, region })
      .then(response => {
        setRecommendations(response.data);
      })
      .catch(error => {
        alert('Failed to fetch recommendations');
      });
  };

  return (
    <div>
      <h1>Recommendations</h1>
      <input type="text" placeholder="Soil Type" onChange={(e) => setSoilType(e.target.value)} />
      <input type="text" placeholder="Crop Variety" onChange={(e) => setCropVariety(e.target.value)} />
      <input type="text" placeholder="Region" onChange={(e) => setRegion(e.target.value)} />
      <button onClick={getRecommendations}>Get Recommendations</button>
      <div>
        {recommendations.map((rec, index) => (
          <div key={index}>
            <h3>{rec.name}</h3>
            <p>Dosage: {rec.dosage}</p>
            <p>Application Method: {rec.application_method}</p>
          </div>
        ))}
      </div>
    </div>
  );
}

export default Recommendation;
